addons/controllers/cpi/vspherecpiconfig_utils.go

```diff
@@ -216,6 +219,44 @@ func (r *VSphereCPIConfigReconciler) mapCPIConfigToDataValues(ctx context.Contex
 	return nil, errors.Errorf("Invalid CPI mode %s, must either be %s or %s", cpiConfig.Spec.VSphereCPI.Mode, VSphereCPIParavirtualMode, VsphereCPINonParavirtualMode)
 }
 
+// mapCPIConfigToProviderServiceAccount maps CPIConfig and cluster to the corresponding service account
+func (r *VSphereCPIConfigReconciler) mapCPIConfigToProviderServiceAccount(_ context.Context, _ *cpiv1alpha1.VSphereCPIConfig, cluster *clusterapiv1beta1.Cluster) *capvvmwarev1beta1.ProviderServiceAccount {
+	serviceAccount := &capvvmwarev1beta1.ProviderServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getCCMName(cluster),
+			Namespace: cluster.Namespace,
+		},
+		Spec: capvvmwarev1beta1.ProviderServiceAccountSpec{
+			Ref: &v1.ObjectReference{Name: cluster.Name, Namespace: cluster.Namespace},
+			Rules: []rbacv1.PolicyRule{
+				{
+					Verbs:     []string{"get", "create", "update", "patch", "delete"},
+					APIGroups: []string{"vmoperator.vmware.com"},
+					Resources: []string{"virtualmachineservices", "virtualmachineservices/status"},
+				},
+				{
+					Verbs:     []string{"get", "list"},
+					APIGroups: []string{"vmoperator.vmware.com"},
+					Resources: []string{"virtualmachines", "virtualmachines/status"},
+				},
+				{
+					Verbs:     []string{"get", "create", "update", "list", "patch", "delete", "watch"},
+					APIGroups: []string{"nsx.vmware.com"},
+					Resources: []string{"ippools", "ippools/status"},
+				},
+				{
+					Verbs:     []string{"get", "create", "update", "list", "patch", "delete"},
+					APIGroups: []string{"nsx.vmware.com"},
+					Resources: []string{"routesets", "routesets/status"},
+				},
+			},
+			TargetNamespace:  "vmware-system-cloud-provider",
```

Comment:
Should be constants. Additionally, is there a case where the namespace needs to be configurable?

Reply:
changed to constants. Yes, the namespace should be configured the same one as the namespace where cloud-control-manager is deployed.